rules/how-to-easily-start-a-react-project/rule.md

```diff
@@ -1,38 +1,77 @@
 ---
 type: rule
-archivedreason: 
 title: Do you know how to easily start a React project?
-guid: 7284ecb9-2844-42ef-9283-0fb14439e182
 uri: how-to-easily-start-a-react-project
-created: 2019-04-29T20:17:31.0000000Z
 authors:
-- title: Shane Ye
-  url: https://ssw.com.au/people/shane-ye
+  - title: Shane Ye
+    url: https://ssw.com.au/people/shane-ye
+  - title: Harry Ross
+    url: https://ssw.com.au/people/harry-ross
 related: []
 redirects:
-- do-you-know-how-to-easily-start-a-react-project
-
+  - do-you-know-how-to-easily-start-a-react-project
+created: 2019-04-29T20:17:31.000Z
+archivedreason: null
+guid: 7284ecb9-2844-42ef-9283-0fb14439e182
 ---
+## Vite
+
+Vite is a reliable frontend build tool for building fast and optimised frontend web apps that has easy integration with a wide range of frontend web frameworks, and built-in Typescript support.  
+
+Vite is much faster than using create-react-app, mainly because Vite does not use Webpack for bundling assets. It instead uses esbuild and Rollup, which are much faster and more modern tools for building great frontend web apps. 
+
+::: good
+![Good example: Use Vite to generate a react-ts project](vite.png)
+:::
+**Note:** Vite requires Node version 14.18+ or 16+. 
+
+1. Run:
+
+```shell
+npm create vite@latest 
+```
 
-Create-React-App is a boilerplate provided by Facebook for almost any React project. Create React App is an officially supported way to create single-page React applications. It offers a modern build setup with no configuration.
+2. Enter the name of your project
+3. Select "React" from the list of frameworks
+
+![Figure: The framework options with Vite](vite-pick.png)
+
+4. Then, select the "TypeScript" variant from the list
+
+![Figure: The variant options with Vite](vite-pick-typescript.png)
+
+5. All done! Now navigate into the directory and run the app with the following commands:
+
+```bash
+cd {{ PROJECT_NAME }}
+npm install 
+npm run dev
+```
+## Create React App
```

Comment:
Let's just delete "Create React App" section, or mark this as bad practice 😄

Reply:
Done